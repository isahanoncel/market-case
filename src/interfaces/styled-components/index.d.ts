// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft';

export type Colors =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black100'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
