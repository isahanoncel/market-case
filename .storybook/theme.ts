import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
