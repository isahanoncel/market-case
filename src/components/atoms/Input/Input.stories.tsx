import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Input from '.';

export default {
  title: 'Atoms/Input',
  component: Input,
} as ComponentMeta<typeof Input>;

export const Default: ComponentStory<typeof Input> = (args) => (
  <Input {...args} />
);
