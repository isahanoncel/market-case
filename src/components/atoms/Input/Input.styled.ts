import styled from 'styled-components';

export const StyledInputWrapper = styled.div`
  margin-bottom: 17px;
`;
export const StyledInput = styled.input`
  font-size: 14px;
  outline: none;
  padding: 12px 16px;
  color: ${({ theme }) => theme.colors.black600};
  border: 2px solid ${({ theme }) => theme.colors.black100};
  border-radius: 2px;

  &::placeholder {
    color: ${({ theme }) => theme.colors.black100};
  }
`;
