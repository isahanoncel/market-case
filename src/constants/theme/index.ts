import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
