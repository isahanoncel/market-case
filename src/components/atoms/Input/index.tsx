import React, { FC } from "react";
import Flex from "../Flex";
import Label from "../Label";
import * as S from "./Input.styled";

interface IInput {
  type?: string;
  placeholder: string;
  onChange:(value:string) => void;
}
const Input: FC<IInput> = ({ type = "text", placeholder,onChange }) => {
  return (
    <S.StyledInputWrapper>
      <Flex flexDirection="column">
        <Label size={14} weight={400} color="black100">
        {placeholder}
        </Label>
        <S.StyledInput onChange={(e) => onChange(e.target.value)} type={type} />
      </Flex>
    </S.StyledInputWrapper>
  );
};

export default Input;
