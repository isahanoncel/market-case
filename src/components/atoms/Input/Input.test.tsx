import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Input from '.';
import theme from '../../../constants/theme';

test('Input snapshot testing', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Input onChange={() => {}} placeholder='Search...' />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
