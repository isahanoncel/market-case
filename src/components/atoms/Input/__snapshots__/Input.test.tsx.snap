// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot testing 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  margin-bottom: 17px;
}

.c2 {
  font-size: 14px;
  outline: none;
  padding: 12px 16px;
  color: #525252;
  border: 2px solid #E0E0E0;
  border-radius: 2px;
}

.c2::-webkit-input-placeholder {
  color: #E0E0E0;
}

.c2::-moz-placeholder {
  color: #E0E0E0;
}

.c2:-ms-input-placeholder {
  color: #E0E0E0;
}

.c2::placeholder {
  color: #E0E0E0;
}

@media (min-width:0px) and (max-width:768px) {
  .c1 {
    height: auto;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <input
      className="c2"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
    />
  </div>
</div>
`;
